### PR TITLE
Proposal of a fix for bug #1742

### PR DESCRIPTION
**Pull request by hippo91**
### Fixes / new features
- This patch is a proposal that fixes the bug #1742.
- It consists in avoiding the call to the ``MessagesHandlerMixin.disable`` method when the line is a commented line.

### CODE REVIEW

**Comment by hippo91** [comment]
I seems that Travis CI build failed due to a connection error. How can i launch again the CI tests?

**Comment by brycepg** [comment]
I just open and close the issue or push to the pull request github branch to restart the CI process

**Comment by hippo91** [comment]
Thanks @brycepg ! I did what you suggested.
I thought there was a smarter way to ask for the CI to be launched again but it doesn't seem to be the case.

File: pylint/lint.py [hunk 1]
```diff
@@ -674,7 +674,10 @@ def process_tokens(self, tokens):
                             self.add_message('file-ignored', line=start[0])
                             self._ignore_file = True
                             return
-                        meth(msgid, 'module', start[0])
+                        sharp_list = content.split('#')
```
Comment (PCManticore) [inline pylint/lint.py]: I am not entirely sure I understand this, can you simplify the condition by splitting it into multiple clauses and assign variables names to each? Also, would this disable the following lines as well, if I understand correctly from your comments? I don't think we should do that, if that is the case.

**Comment by hippo91** [comment]
@PCManticore i made the change you requested. Tell me if it's ok. 
I can't understand the problem with the CI. One unit test failed, telling me a 'line-too-long' message is emitted for line 37 in ``line_too_long.py``. It's normal and i added this information in ``line_too_long.txt`` but it is as if tox does not notice it.

**Comment by PCManticore** [comment]
@hippo91 can you rebase your branch? It contains too many changes from master

**Comment by hippo91** [comment]
@PCManticore I rebased my branch (``git rebase upstream/master``) and on my desktop if i launch:
``git diff master/bug_1742 upstream/master`` only six files are different.
But here github shows 125 files. What did i miss?

**Comment by PCManticore** [comment]
@hippo91 I'm not exactly sure without pulling your branch and checking it. I would suggest to pull master locally and then do a rebase on top of your branch. You might also take a look at `git reflog` to see where this diverged into this state you're seeing here. anyway, I'll take another look next week!

**Comment by hippo91** [comment]
@PCManticore don't bother with this PR. I'll try what you said but if i can't succeed i'll make a fresh new PR.

**Comment by hippo91** [comment]
@PCManticore it's ok now, only six files are different from master.
However i'm still facing the problem of a unittest failing with:
 ``Unexpected in testdata``
``37: line-too-long``
whereas i explicitly put in ``line_too_long.txt`` that this message should by emit on line 37.
Have you got any idea?

**Comment by PCManticore** [comment]
Thanks for updating the PR @hippo91 Sorry for not commenting earlier, I missed the comment. From what I see, you don't have any pragma on line 37. I think it makes sense to emit on line 37 as far as I can see, since the line is indeed too long. OR where you expecting not to be like this due to the pragma on line 35?

**Comment by hippo91** [comment]
@PCManticore thank you so much!
I'm so stupid to have forgotten this pragma!

**Comment by hippo91** [comment]
@PCManticore  do I need to backport this PR to 1.8 branch?

**Pull request merged**